lib/icingadb/icingadb.hpp

```diff
@@ -232,6 +233,8 @@ class IcingaDB : public ObjectImpl<IcingaDB>
 	// initialization, the value is read-only and can be accessed without further synchronization.
 	static String m_EnvironmentId;
 	static std::mutex m_EnvironmentIdInitMutex;
+
+	static std::unordered_set<Type*> m_IndexedTypes;
```

Comment:
Why did you call it "indexed"?

Reply:
GetTypes() produces a vector, lookup: O(n). In contrast that is a hash table, lookup: O(1).

Reply:
Why can't this be a `std::unordered_set<Type::Ptr>` though? Compiled fine in my test, is this something about Boost version compatibility?

Reply:
It simply doesn’t have to ensure even that the pointers are valid, it's lookup-only.

Reply:
We usually just use `T::Ptr` instead of discussing where just `T*` would also be fine, so I don't see why this place should be special. Yes, it's fine here as the `Type` objects in here should be kept alive forever by some other reference, but in general, it might happen that some object in the set is freed, then a new object is allocated at the same address and then it would be in the set when it shouldn't.

Reply:
Just both instances of `std::unordered_set<Type*>`. But should probably also remove a few `.get()`, otherwise it would pointlessly increment and decrement reference counts.

Reply:
Boost version?

Reply:
1.79.0

Reply:
Please try 1.66.

Reply:
I don't have that version lying around locally, so please let the CI do its job.

Reply:
Well, that looks like a nope to me:
> error: static assertion failed: hash function must be invocable with an argument of key type

Reply:
Shall I undo?